chingu-voyages/v50-tier2-team-14
# Leaflet map

# Description
Implemented a drop-down menu that allows users to select restaurants by state. When the user clicks the "Search" button, they are directed to another page that displays the restaurants with markers on a map. The markers are automatically centered in the middle of the map. When users hover over the markers, the name of the restaurant is displayed. On top of the map, it has categories like Steak, Burger, Dessert, Pizza, and Fried Chicken. When the user clicks on the image, it filters out the restaurants located in that specific state and belongs to the selected category.

## Main changes explained:

- created Map directory included FilterByCategory.js, FilterState.js, and RestaurantByState.js
- added drop-down menu for State selection
- added category images to filter restaurants by category
- installed react-router, leaflet-react, leaflet, and axios
- created a map to display the locations of restaurants.

## How to test:
1. check into current branch
2. do `npm install` and `npm start` to run this PR locally
3. go to localhost:3000
4. select a state from drop-down menu and click 'Search' button
5. hover to any restaurants on the map and click on it


## Screenshots or videos of changes:
![Screenshot 2024-08-08 at 1 15 10 AM](https://github.com/user-attachments/assets/594209b6-d395-4687-8526-955e9489c7a1)

![Screenshot 2024-08-08 at 1 12 35 AM](https://github.com/user-attachments/assets/2ce09af5-72c4-487d-a7f4-4c6a39257657)




## Note:
Need to improve the interface - The color for the drop-down menu looks different from Figma.
